### PR TITLE
lint: cull most dead code from akash repository

### PR DESCRIPTION
**Pull request by Ropes**
There are two places where removing dead/unused code that the linter
found was problematic:
 * cmd/akashctl/provider.go: Staged command for running the provider.
   * Resolved with a simple unused variable call to the single function until we're ready to use the command.
 * pkg/client/clientset/versioned/fake/register.go:32:5:
 `parameterCodec` is unused (deadcode) ... kubernetes generated code that will always be generated.
   * This has been resolved with lint --skip-dir flag for `pkg/client/clientset` k8s client directory.

Otherwise all deadcode removed.
fixes: #539

### CODE REVIEW

File: cmd/akashctl/provider.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	pmodule "github.com/ovrclk/akash/x/provider"
 )
 
+var _ = providerCmd(codec.New()) // Disables deadcode lint check until provderCmd is ready for use.
```
Comment (boz) [inline cmd/akashctl/provider.go:27]: :+1:

File: provider/cluster/kube/client_test.go [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
-	"github.com/tendermint/tendermint/libs/log"
-)
-
+/* Integration test stub
 func kubeClient(t *testing.T) Client {
 	client, err := NewClient(log.NewTMLogger(os.Stdout), "host", strings.ToLower(t.Name()))
 	assert.NoError(t, err)
 	return client
 }
+*/
```
Comment (boz) [inline provider/cluster/kube/client_test.go:9]: meh, `rm -rf` :shrug:

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -48,8 +48,8 @@ jobs:
       - name: lint all
         run: $(go env GOPATH)/bin/golangci-lint run ./... --issues-exit-code=0
       - name: lint deadcode
-        run: $(go env GOPATH)/bin/golangci-lint run ./... --no-config --disable-all --enable deadcode
-        continue-on-error: true
+        run: $(go env GOPATH)/bin/golangci-lint run ./... --no-config  --disable-all --enable deadcode --skip-dirs=pkg/client/clientset 
+        continue-on-error: false
```
Comment (boz) [inline .github/workflows/tests.yml:52]: mind adding those codegen dirs to `.golangci.yaml` too?

Comment (Ropes) [reply .github/workflows/tests.yml:52]: sure!

File: .golangci.yaml [hunk 4]
```diff
@@ -5,3 +5,10 @@ issues:
   exclude:
     - comment on exported (method|function|type|const|var)
   exclude-use-default: true
+
+# Skip generated k8s code
+skip-dirs:
+  - pkg/client/clientset
+
+# Skip vendor/ etc
+skip-dirs-use-default: true
```
Comment (troian) [inline .golangci.yaml:14]: Newline is missing

Comment (Ropes) [reply .golangci.yaml:14]: YAML :facepalm: thanks!

**Pull request merged**